### PR TITLE
[osg] Fix can't find sdl header files.

### LINKED ISSUE #30289: [osg] Build error
Package: osg[core,examples,freetype]:x64-windows -> 3.6.5#18

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.35.32215.0
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: ffe7360e0 2023-03-17 (2 days ago)

**To Reproduce**

`vcpkg install `

**Failure logs**

```
-- Downloading https://github.com/openscenegraph/OpenSceneGraph/archive/OpenSceneGraph-3.6.5.tar.gz -> openscenegraph-OpenSceneGraph-OpenSceneGraph-3.6.5.tar.gz...
-- Extracting source C:/Users/LeanderSchulten/git_projects/vcpkg/downloads/openscenegraph-OpenSceneGraph-OpenSceneGraph-3.6.5.tar.gz
-- Applying patch link-libraries.patch
-- Applying patch collada.patch
-- Applying patch fix-sdl.patch
-- Applying patch fix-nvtt-squish.patch
-- Applying patch plugin-pdb-install.patch
-- Applying patch use-boost-asio.patch
-- Applying patch osgdb_zip_nozip.patch
-- Applying patch openexr3.patch
-- Applying patch unofficial-export.patch
-- Using source at C:/Users/LeanderSchulten/git_projects/vcpkg/buildtrees/osg/src/raph-3.6.5-41fa3cfd98.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: C:/Users/LeanderSchulten/git_projects/vcpkg/downloads/tools/cmake-3.25.1-windows/cmake-3.25.1-windows-i386/bin/cmake.exe --build . --config Release --target install -- -v -j9
    Working Directory: C:/Users/LeanderSchulten/git_projects/vcpkg/buildtrees/osg/x64-windows-rel
    See logs for more information:
      C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\install-x64-windows-rel-out.log

Call Stack (most recent call first):
  vcpkg_installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  vcpkg_installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/osg/portfile.cmake:132 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)



```
<details><summary>C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\install-x64-windows-rel-out.log</summary>

```
[1/1525] C:/Users/LEANDE~1/GIT_PR~1/vcpkg/DOWNLO~1/tools/CMAKE-~1.1-W/CMAKE-~1.1-W/bin/cmcldeps.exe RC C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\PlatformSpecifics\Windows\OpenThreadsVersionInfo.rc src\OpenThreads\win32\CMakeFiles\OpenThreads.dir\__\__\__\PlatformSpecifics\Windows\OpenThreadsVersionInfo.rc.res.d src\OpenThreads\win32\CMakeFiles\OpenThreads.dir\__\__\__\PlatformSpecifics\Windows\OpenThreadsVersionInfo.rc.res "Note: including file: " "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.35.32215/bin/Hostx64/x64/cl.exe" C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe -DOPENTHREADS_EXPORTS -DOpenThreads_EXPORTS -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0400 -I C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -I C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include -c65001 /DWIN32 /fo src\OpenThreads\win32\CMakeFiles\OpenThreads.dir\__\__\__\PlatformSpecifics\Windows\OpenThreadsVersionInfo.rc.res C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\PlatformSpecifics\Windows\OpenThreadsVersionInfo.rc
[2/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -DOPENTHREADS_EXPORTS -DOpenThreads_EXPORTS -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0400 -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Fosrc\OpenThreads\win32\CMakeFiles\OpenThreads.dir\Win32ThreadBarrier.cpp.obj /Fdsrc\OpenThreads\win32\CMakeFiles\OpenThreads.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\src\OpenThreads\win32\Win32ThreadBarrier.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[3/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -DOPENTHREADS_EXPORTS -DOpenThreads_EXPORTS -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0400 -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Fosrc\OpenThreads\win32\CMakeFiles\OpenThreads.dir\Win32Mutex.cpp.obj /Fdsrc\OpenThreads\win32\CMakeFiles\OpenThreads.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\src\OpenThreads\win32\Win32Mutex.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
...
Skipped 2500 lines
...

[1273/1525] cmd.exe /C "cd . && C:\Users\LeanderSchulten\git_projects\vcpkg\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=examples\osglogicop\CMakeFiles\example_osglogicop.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\link.exe  examples\osglogicop\CMakeFiles\example_osglogicop.dir\osglogicop.cpp.obj  /out:bin\osglogicop.exe /implib:lib\osglogicop.lib /pdb:bin\osglogicop.pdb /version:0.0 /machine:x64 /nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF   /subsystem:console  lib\osgViewer.lib  lib\osgText.lib  lib\osgGA.lib  lib\osgDB.lib  C:\Users\LeanderSchulten\git_projects\vcpkg\vcpkg_installed\x64-windows\lib\zlib.lib  lib\osgUtil.lib  lib\osg.lib  lib\OpenThreads.lib  opengl32.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
[1274/1525] cmd.exe /C "cd . && C:\Users\LeanderSchulten\git_projects\vcpkg\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=examples\osglightpoint\CMakeFiles\example_osglightpoint.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\link.exe  examples\osglightpoint\CMakeFiles\example_osglightpoint.dir\osglightpoint.cpp.obj  /out:bin\osglightpoint.exe /implib:lib\osglightpoint.lib /pdb:bin\osglightpoint.pdb /version:0.0 /machine:x64 /nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF   /subsystem:console  lib\osgViewer.lib  lib\osgSim.lib  lib\osgGA.lib  lib\osgText.lib  lib\osgDB.lib  C:\Users\LeanderSchulten\git_projects\vcpkg\vcpkg_installed\x64-windows\lib\zlib.lib  lib\osgUtil.lib  lib\osg.lib  lib\OpenThreads.lib  opengl32.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
[1275/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osglogo\CMakeFiles\example_osglogo.dir\osglogo.cpp.obj /Fdexamples\osglogo\CMakeFiles\example_osglogo.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osglogo\osglogo.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[1276/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osggpucull\CMakeFiles\example_osggpucull.dir\AggregateGeometryVisitor.cpp.obj /Fdexamples\osggpucull\CMakeFiles\example_osggpucull.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osggpucull\AggregateGeometryVisitor.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[1277/1525] cmd.exe /C "cd . && C:\Users\LeanderSchulten\git_projects\vcpkg\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=examples\osglogo\CMakeFiles\example_osglogo.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\link.exe  examples\osglogo\CMakeFiles\example_osglogo.dir\osglogo.cpp.obj  /out:bin\osglogo.exe /implib:lib\osglogo.lib /pdb:bin\osglogo.pdb /version:0.0 /machine:x64 /nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF   /subsystem:console  lib\osgViewer.lib  lib\osgText.lib  lib\osgGA.lib  lib\osgDB.lib  C:\Users\LeanderSchulten\git_projects\vcpkg\vcpkg_installed\x64-windows\lib\zlib.lib  lib\osgUtil.lib  lib\osg.lib  lib\OpenThreads.lib  opengl32.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
[1278/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -DUSE_SDL2 -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include -external:IC:\Users\LeanderSchulten\git_projects\vcpkg\vcpkg_installed\x64-windows\include\SDL2 -external:IC:\Users\LeanderSchulten\git_projects\vcpkg\vcpkg_installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osgmovie\CMakeFiles\example_osgmovie.dir\osgmovie.cpp.obj /Fdexamples\osgmovie\CMakeFiles\example_osgmovie.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osgmovie\osgmovie.cpp
FAILED: examples/osgmovie/CMakeFiles/example_osgmovie.dir/osgmovie.cpp.obj 
C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -DUSE_SDL2 -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include -external:IC:\Users\LeanderSchulten\git_projects\vcpkg\vcpkg_installed\x64-windows\include\SDL2 -external:IC:\Users\LeanderSchulten\git_projects\vcpkg\vcpkg_installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osgmovie\CMakeFiles\example_osgmovie.dir\osgmovie.cpp.obj /Fdexamples\osgmovie\CMakeFiles\example_osgmovie.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osgmovie\osgmovie.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osgmovie\osgmovie.cpp(694): fatal error C1083: Cannot open include file: 'SDL/SDL.h': No such file or directory
[1279/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osggraphicscost\CMakeFiles\example_osggraphicscost.dir\osggraphicscost.cpp.obj /Fdexamples\osggraphicscost\CMakeFiles\example_osggraphicscost.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osggraphicscost\osggraphicscost.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[1280/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osgimpostor\CMakeFiles\example_osgimpostor.dir\TestManipulator.cpp.obj /Fdexamples\osgimpostor\CMakeFiles\example_osgimpostor.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osgimpostor\TestManipulator.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[1281/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osggpx\CMakeFiles\example_osggpx.dir\osggpx.cpp.obj /Fdexamples\osggpx\CMakeFiles\example_osggpx.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osggpx\osggpx.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[1282/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osgimpostor\CMakeFiles\example_osgimpostor.dir\osgimpostor.cpp.obj /Fdexamples\osgimpostor\CMakeFiles\example_osgimpostor.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osgimpostor\osgimpostor.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[1283/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osgmanipulator\CMakeFiles\example_osgmanipulator.dir\osgmanipulator.cpp.obj /Fdexamples\osgmanipulator\CMakeFiles\example_osgmanipulator.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osgmanipulator\osgmanipulator.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[1284/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osggpucull\CMakeFiles\example_osggpucull.dir\osggpucull.cpp.obj /Fdexamples\osggpucull\CMakeFiles\example_osggpucull.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osggpucull\osggpucull.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[1285/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osgmultiplerendertargets\CMakeFiles\example_osgmultiplerendertargets.dir\osgmultiplerendertargets.cpp.obj /Fdexamples\osgmultiplerendertargets\CMakeFiles\example_osgmultiplerendertargets.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osgmultiplerendertargets\osgmultiplerendertargets.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[1286/1525] C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\x64-windows-rel\include -IC:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /W4 /wd4589 /wd4706 /wd4127 /wd4100 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Foexamples\osgmultiplemovies\CMakeFiles\example_osgmultiplemovies.dir\osgmultiplemovies.cpp.obj /Fdexamples\osgmultiplemovies\CMakeFiles\example_osgmultiplemovies.dir\ /FS -c C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osgmultiplemovies\osgmultiplemovies.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "name": "lichtsteuerung",
  "version-string": "2.0.0",
  "dependencies": [
    {
      "name": "osg",
      "features": [
        "examples"
      ],
      "default-features": false
    }
  ]
}

```
</details>

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/30289
Add the dependency library sdl1 in the examples feature.
Compile test pass with following triplets:
```
x86-windows
x64-windows
x64-windows-static
```
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/osg/portfile.cmake [hunk 1]
```diff
@@ -165,6 +166,6 @@ if(NOT VCPKG_BUILD_TYPE)
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/openscenegraph.pc" "\\\n" " ")
 endif()
 vcpkg_fixup_pkgconfig()
-
+set(osg_OPENGL_PROFILE GL3)
```
Comment (dg0yt) [inline ports/osg/portfile.cmake:169]: This line has no effect.

File: ports/osg/portfile.cmake [hunk 2]
```diff
@@ -138,6 +138,7 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     file(APPEND "${CURRENT_PACKAGES_DIR}/include/osg/Config" "#ifndef OSG_LIBRARY_STATIC\n#define OSG_LIBRARY_STATIC 1\n#endif\n")
 endif()
 
+set(tools osg2cpp osgshaderpipeline)
```
Comment (dg0yt) [inline ports/osg/portfile.cmake:141]: This variable gets overwritten when building with `tools`.
And do these tool need be built without `tools` feature?

File: ports/osg/vcpkg.json [hunk 3]
```diff
@@ -93,6 +93,7 @@
             "freetype"
           ]
         },
+        "sdl1",
```
Comment (dg0yt) [inline ports/osg/vcpkg.json:96]: My conclusion from https://github.com/microsoft/vcpkg/issues/30289#issuecomment-1486125987 is that we need a dependency on `osg[sdl1]`, not on `sdl1`.

File: ports/osg/portfile.cmake (hunk 2 above)
Comment (jimwang118) [reply ports/osg/portfile.cmake:141]: Yes, these two tools will also build without the tools feature.

File: ports/osg/vcpkg.json (hunk 3 above)
Comment (jimwang118) [reply ports/osg/vcpkg.json:96]: The error date is that `SDL/SDL.h` cannot be found, and the header file path after sdl1 installation is this path.
`C:\Users\LeanderSchulten\git_projects\vcpkg\buildtrees\osg\src\raph-3.6.5-41fa3cfd98.clean\examples\osgmovie\osgmovie.cpp(694): fatal error C1083: Cannot open include file: 'SDL/SDL.h': No such file or directory`

**Comment by dg0yt** [comment]
https://github.com/microsoft/vcpkg/pull/32313#discussion_r1247449964 is not resolved. The `sdl1` feature is to control everything about needing and using sdl1.

**Comment by dg0yt** [comment]
Invalid, cf. https://github.com/microsoft/vcpkg/issues/30289#issuecomment-1615919559

**Comment by dg0yt** [comment]
Actual fix: https://github.com/microsoft/vcpkg/pull/32335/commits/a32ddfe44d870c6d0405e578ef1703fa13574785

**Comment by jimwang118** [comment]
Duplicate of [32335](https://github.com/microsoft/vcpkg/pull/32335)

**Pull request closed without merging**